src/javascript/_common/base/livechat.js

```diff
@@ -66,9 +79,75 @@ const LiveChat = (() => {
         
     };
 
+    // Called when logging out to end ongoing chats if there is any
+    const endLiveChat = () => new Promise ((resolve) => {
+        session_variables = { loginid: '', landing_company_shortcode: '', currency: '', residence: '', email: '' };
+        window.LiveChatWidget.call('set_session_variables', session_variables);
+        window.LiveChatWidget.call('set_customer_email', ' ');
+        window.LiveChatWidget.call('set_customer_name', ' ');
+        
+        try {
+            const customerSDK = init({
+                licenseId: licenseID,
+                clientId : clientID,
+            });
+            customerSDK.on('connected', () => {
+                if (window.LiveChatWidget.get('chat_data')) {
+                    const { chatId, threadId } = window.LiveChatWidget.get('chat_data');
+                    if (threadId) {
+                        customerSDK.deactivateChat({ chatId });
+                    }
+                }
+                resolve();
+            });
+        } catch (e){
+            resolve();
+        }
+
+    });
+
+    // Delete existing LiveChat instance when there is no chat running
+    const livechatDeletion = () => new Promise ((resolve) => {
+        if (window.LiveChatWidget){
+            window.LiveChatWidget.on('ready', () => {
+                try {
+                    if (window.LiveChatWidget.get('customer_data').status !== 'chatting') {
+                        window.LiveChatWidget.call('destroy');
+                        resolve();
+                    }
+                } catch (e) {
+                    resolve();
```

Comment:
reject(e)?

Reply:
I don't think I'll have to reject it, just resolve will do

Reply:
We need the resolve just to make sure no errors are logged on the console. These errors pop up when user opens certain links that causes the url to change more than once or when user travels between pages too quickly (on either occasions we can ignore the error since the rerouting function will be called again)